scripts/generate-icon-exports.js

```diff
@@ -9,101 +9,161 @@ const { stripIndents } = require('common-tags');
+  '../src/components/internals/icons/index.js'
+);
 
-const prettierConfig = rcfile('prettier');
+const publicIconsPath = path.join(
+  __dirname,
+  '../src/components/icons/svg/*.svg'
+);
 
-glob(importPath, async (err, files) => {
-  files.forEach(fileName => {
-    const fileNameWithoutExtension = path.basename(fileName, iconFileExt);
-    const componentName = upperFirst(camelCase(fileNameWithoutExtension));
-
-    const svgCode = fs.readFileSync(fileName, 'UTF-8');
-    const jsCode = svgr.sync(
-      svgCode,
-      {
-        icon: false,
-        svgoConfig: {
-          plugins: [
-            { removeViewBox: false },
-            { prefixIds: true },
-            // same result as rollup plugin
-            { prefixIds: { prefix: fileNameWithoutExtension } },
-
-            // Keeps ID's of svgs so they can be targeted with CSS
-            { cleanupIDs: false },
-          ],
-        },
-        // same as the rollup plugin
-        plugins: ['@svgr/plugin-svgo', '@svgr/plugin-jsx'],
-      },
-      { componentName }
-    );
+const publicRawExportsPath = path.join(
+  __dirname,
+  '../src/components/icons/raw-components/'
+);
 
-    fs.writeFileSync(
-      path.join(rawExportsPath, `/${fileNameWithoutExtension}.js`),
-      prettier.format(jsCode, prettierConfig)
-    );
+const publicIconComponentsPath = path.join(
+  __dirname,
+  '../src/components/icons/components/'
+);
 
-    const displayName = `${componentName}Icon`;
+const publicIndexPath = path.join(
+  __dirname,
+  '../src/components/icons/index.js'
+);
 
-    const importStatement = stripIndents`
-      import React from 'react';
-      import { getIconStyles, iconPropTypes } from '../create-styled-icon';
-      import ${componentName} from '../raw-components/${fileNameWithoutExtension}';
+const createStyledIconPath = path.join(
+  __dirname,
+  '../src/components/internals/icons/create-styled-icon'
+);
 
-      const Component = props => <${componentName} {...props} css={theme => getIconStyles(props, theme)} />;
+shelljs.exec(
+  `rm -rf ${publicRawExportsPath} ${publicIconComponentsPath} ${internalRawExportsPath} ${internalIconComponentsPath}`
+);
 
-      Component.displayName = '${displayName}';
+shelljs.exec(
+  `mkdir ${publicRawExportsPath} ${publicIconComponentsPath} ${internalRawExportsPath} ${internalIconComponentsPath}`
+);
 
-      // we do this to enable treeshaking
-      // please see https://github.com/alex996/react-css-spinners/issues/1
-      if (process.env.NODE_ENV !== "production") {
-        Component.propTypes = iconPropTypes;
-      }
+const iconsToProcess = [
+  {
+    indexPath: publicIndexPath,
+    svgPath: publicIconsPath,
+    rawExportsPath: publicRawExportsPath,
+    iconComponentsPath: publicIconComponentsPath,
+  },
+  {
+    indexPath: internalIndexPath,
+    svgPath: internalIconsPath,
+    rawExportsPath: internalRawExportsPath,
+    iconComponentsPath: internalIconComponentsPath,
+  },
+];
+
+iconsToProcess.forEach(data => {
```

Comment:
now we go through two "rounds".

first time for the public icons, second for the internal.